Source/WebKit/Platform/IPC/StreamServerConnection.h

```diff
@@ -154,18 +163,20 @@ void StreamServerConnection::sendAsyncReply(AsyncReplyID asyncReplyID, Arguments
     m_connection->sendSyncReply(WTFMove(encoder));
 }
 
-inline void StreamServerConnection::Handle::encode(Encoder& encoder) const
+inline void StreamServerConnection::Handle::encode(Encoder& encoder)
 {
-    encoder << outOfStreamConnection << buffer;
+    encoder << WTFMove(outOfStreamConnection) << buffer;
```

Comment:
This leaves the instance in inconsistent state behind the callers back.
So I think part of the work in this patch is to ensure rvalue-only encoder functions work.
E.g. fix until `void encode(Encoder&) &&;` works...

Reply:
Oh that's a good point! It does work already, just need to add the `&&` ref-qualifier.